rfcs/text/0000-rfc-library-preparation.md

```diff
@@ -0,0 +1,423 @@
+
+### Challenges
+
+The current implementation for representing sequence library preparations in the HCA DCP creates challenges for both HCA data contributors and consumers. Some of those challenges include:
+
+1. Data contributors are not required to indicate which data files represent sequencing the same library preparation more than once. Being able to do this is necessary to accurately represent the experimental design.
+1. Data consumers find it challenging to determine which data files are from the same library preparation and therefore must be processed together. All data generated from one sequencing library preparation must be processed together. 
+1. Data consumers find it challenging to understand the relationship between cell suspensions from plate-based and droplet-based sequencing technologies because of granularity differences (1 cell versus many cells). This inconsistency creates confusion for data consumers when they are browsing or searching for data.
+
+### User Stories
+
+As a data contributor, I would like to be able to indicate which data files were generated from my library preparations so that I can represent my experiment accurately.
+
+As a data consumer, I would like to be able to identify which data files are from the same library preparation so that I can process all sequencing files which belong together at the same time.
+
+As a data consumer, I would like to be confident that data files from the same library preparation were processed together so that I know the expression matrices are accurate.
+
+## Scientific "guardrails" [optional]
+
+*Describe recommended or mandated review from HCA Science governance to ensure that the RFC addresses the needs of the scientific community.*
+
+TBD
+
+## Detailed Design
+
+### Overview
+
+We will resolve the challenges described above by **creating a first-class library preparation biomaterial entity** as part of the HCA metadata standard. This approach will allow the metadata model to explicitly represent which sequencing data files came from the same library preparation even if the data files were generated across multiple sequencing runs (i.e. multiple processes). This approach would also enable data consumers to more easily identify all data files associated with the same library preparation (i.e. from one logical unit) which is required for correct data processing. Unless and until the fundamental way that DNA sequencing experiments are performed changes, all sequencing-based experiments will involve generating library preparations; therefore, the addition of this entity to the metadata model will be used by all current and future sequence-based datasets.
+
```

Comment:
I find it concerning that we are solving a downstream processing problem by changing the storage structure of the data.  The lack of any separation between storage and processing is severely limiting.   This is not sustainable if we have other or conflicting analysis needs that require being notified in a different manner.

This is the only short-term hack currently available, so I am not suggesting that it not be done, but that it shouldn't be considered a general solutuon.

It also creates complexity as it is creating bundles that look different to the user.

Reply:
> it is creating bundles that look different to the user.

While I agree that implementation of this RFC means technically bundles might look different to the user (some bundles might have a single triplet of fastq files if an LP was sequenced once, while other bundles might have multiple triplets of fastq files if an LP was sequenced more than once), the overwhelming opinion I get from the comp bios I've talked to (e.g. @barkasn @kishorikonwar) is that "all fastq files from the same library preparation need to be processed together", not just in the DCP but as a general rule for processing single cell sequencing data anywhere.

Thus, from my perspective, providing an efficient way to get all fastq files per LP is preferred over providing that exact same structure within a bundle (e.g. 1 triplet per bundle, what we've been doing up until now). We already know the structure within a bundle is going to be different for imaging data, anyway.

Reply:
I went ahead and created a [scratch DCP terminology page](https://github.com/HumanCellAtlas/dcp/wiki/DCP-Terminology). Here I've written that a bundle is just a collection of file references so if you don't agree please edit/comment :)

Reply:
We also have the Bundle Definition RFC! https://github.com/HumanCellAtlas/dcp-community/pull/93

Reply:
I doubt that an rfc was the right mechanism for that. Why is andrey the only reviewer?

Reply:
And in the latest part of this comedy it turns out there is already [a Google doc](https://docs.google.com/document/d/1DmSoUxfblKMZTb7fkI7mQ0a4dHHXpTTNJ4Ho7Fm9hDY/edit#)

Reply:
@justincc I believe that RFC is still in Draft mode. Re. google docs: we should be slowly migrating the important info from google docs to RFCs to avoid the google drive morass.

Reply:
This isn't suitable for an RFC since it's something that changes in increments over time, You can't start an RFC every time you want to add a definition - that just means no one every writes anything because the process barrier is too high.